src/Compilers/CSharp/Portable/SymbolDisplay/ObjectDisplay.cs

```diff
@@ -184,6 +183,12 @@ private static bool TryReplaceChar(char c, out string replaceWith)
                 case '\v':
                     replaceWith = "\\v";
                     break;
+                case '\u2028':
+                    // U+2028 "LINE SEPARATOR" should be classified by CharUnicodeInfo.GetUnicodeCategory as
+                    // UnicodeCategory.LineSeparator but due to a bug is incorrectly categorized as
+                    // UnicodeCategory.Format. See https://github.com/dotnet/coreclr/issues/3542
+                    replaceWith = "\\u2028";
```

Comment:
There are a lot of calls to CharUnicodeInfo.GetUnicodeCategory in our codebase and some of those sites can still suffer from the bug in this API. For example (in src\Compilers\Core\Portable\UnicodeCharacterUtilities.cs):
```
            UnicodeCategory cat = CharUnicodeInfo.GetUnicodeCategory(ch);
            return IsLetterChar(cat)
                || IsDecimalDigitChar(cat)
                || IsConnectingChar(cat)
                || IsCombiningChar(cat)
                || IsFormattingChar(cat);
``` 

Instead of patching this particular call site in this manner, we should consider creating a wrapper for CharUnicodeInfo.GetUnicodeCategory, which would make necessary adjustments to the value it returns and simply change all call sites to use that wrapper.